hypershift-operator/controllers/hostedcluster/hostedcluster_webhook.go

```diff
@@ -105,53 +138,112 @@ func validateNetworkCIDRs(hc *hyperv1.HostedCluster) field.ErrorList {
 		return errs
 	}
 
-	return compareCIDREntries(cidrEntries)
+	return validateNoCIDROverlap(cidrEntries)
 }
 
 func validateSliceNetworkCIDRs(hc *hyperv1.HostedCluster) field.ErrorList {
 	var cidrEntries []cidrEntry
 
 	for _, cidr := range hc.Spec.Networking.MachineNetwork {
-		ce := cidrEntry{(net.IPNet)(cidr.CIDR), *field.NewPath("spec.networking.MachineNetwork")}
+		ce := cidrEntry{(net.IPNet)(cidr.CIDR), *field.NewPath("hostedcluster.spec.networking.machineNetwork")}
```

Comment:
This will only be called for hostedclusters so it will only be visible when updating one so prefixing it with `hostedcluster` here seems a bit redundant to me. What was the motivation for this?

Reply:
I was doing it for the immutable checks as well.  It's fine here.. we could leave them out.  It's only really important when validating against the cluster network service CIDR.